src/test/java/com/example/music_player/service/DecoratorQueueServiceTest.java

```diff
@@ -0,0 +1,128 @@
+package com.example.music_player.service;
+
+import com.example.music_player.MusicPlayerApplication;
+import com.example.music_player.entity.Source;
+import org.apache.catalina.Server;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.jms.core.JmsTemplate;
+import org.springframework.test.context.ContextConfiguration;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.when;
+
+@AutoConfigureMockMvc
+@ContextConfiguration(classes = MusicPlayerApplication.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, classes = Server.class)
+class DecoratorQueueServiceTest {
```

Comment:
No need to make this unit test to leverage on spring. Mockito extension takes care about mock creation and AlbumService can be instantiated manually without spring ioc container. Moving away from spring utilization in tests make them to execute faster and usually it is considered as best practice.